Finalize a number of deprecations for JAX v0.7.0

jakevdp opened the pull request:
This PR deliberately leaves type registrations untouched; I'll remove those in a followup after ensuring their removal doesn't lead to build breakages.

yashk2810 commented on jax/sharding.py:
```diff
@@ -36,18 +36,18 @@
         (
-            "jax.sharding.PositionalSharding is deprecated. Use"
-            " jax.NamedSharding instead."
+            "jax.sharding.PositionalSharding was deprecated in JAX v0.6.0 and removed"
```
Can you leave this for me? I am working on a CL to remove this as we speak.

jakevdp committed:
Finalize a number of deprecations for JAX v0.7.0

This PR deliberately leaves type registrations untouched; I'll remove those in
a followup after ensuring their removal doesn't lead to build breakages.
